environment/src/main/java/tygronenv/translators/J2PopupData.java

```diff
@@ -1,35 +1,80 @@
 package tygronenv.translators;
 
+import java.util.List;
+
 import eis.eis2java.exception.TranslationException;
 import eis.eis2java.translation.Java2Parameter;
 import eis.eis2java.translation.Translator;
 import eis.iilang.Function;
 import eis.iilang.Identifier;
 import eis.iilang.Numeral;
 import eis.iilang.Parameter;
-import nl.tytech.data.engine.item.Building;
+import eis.iilang.ParameterList;
+import nl.tytech.core.client.event.EventManager;
+import nl.tytech.core.net.serializable.MapLink;
 import nl.tytech.data.engine.item.PopupData;
+import nl.tytech.data.engine.item.SpecialOption;
+import nl.tytech.data.engine.item.SpecialOption.Type;
 
 /**
- * Translate {@link Building} into building(ID, name, [categories], timestate).
- * 
+ * Translate {@link Popup} into request(type, category, contentlinkID,
+ * multipolygon, [visibleStakeholderIDs], [answers]).
  * @author W.Pasman
  *
  */
 public class J2PopupData implements Java2Parameter<PopupData> {
 
+    /**
+     * Translator for additional translations.
+     */
 	private final Translator translator = Translator.getInstance();
 
-	@Override
-	public Parameter[] translate(PopupData b) throws TranslationException {
+	/**
+	 * Empty constructor.
+	 */
+	public J2PopupData() { }
 
-		return new Parameter[] { new Function("request", new Identifier(b.getType().toString()), new Numeral(b.getID()),
-				translator.translate2Parameter(b.getAnswers())[0]) };
+	/**
+	 * Method to translate the PopupData class.
+	 */
+	@Override
+	public Parameter[] translate(final PopupData popup) throws TranslationException {
+	    String typeOfPopup = null;
+	    if (popup.getContentMapLink() == MapLink.SPECIAL_OPTIONS) {
+	        SpecialOption specialOption = EventManager.getItem(MapLink.SPECIAL_OPTIONS, popup.getContentLinkID());
+	        Type optionType = specialOption.getType();
+	        typeOfPopup = optionType.name();
+	    } else if (popup.getContentMapLink() == MapLink.BUILDINGS) {
+	        typeOfPopup = "PERMIT";
+	    } else {
+	        typeOfPopup = "POPUP";
+	    }
+		return new Parameter[] {new Function("request", new Identifier(popup.getType().name()),
+		        new Identifier(typeOfPopup), new Numeral(popup.getID()),
+		        new Numeral(popup.getContentLinkID()),
+		        translator.translate2Parameter(popup.getMultiPolygon())[0],
+				getVisibleForStakeholderIDs(popup.getVisibleForStakeholderIDs()),
```

Comment:
I think the indentation here isn't right. It looks like the 5th parameter is a list because of it. Also maybe put the first parameter on its own line as well?